### PR TITLE
Code Review Tests

### LINKED ISSUE #1: String Errors
There seems to be minor errors in the Strings formats

### PR DESCRIPTION
**Pull request by MaurineM**
- [x] Duplicated some files
- [x] Edited Code

### CODE REVIEW

File: app/src/main/res/duplicate/Fragment.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package com.demo.safeargs
+
```
Comment (Ericgacoki) [inline app/src/main/res/duplicate/Fragment.kt]: Please change the package name to match the parent directory

File: app/src/main/res/duplicate/Fragment.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        binding = FragmentABinding.inflate(layoutInflater)
+       return binding.apply {
+
+           binding?.btnShareData?.setOnClickListener {
+               val person = Person("Maurine", 20)
+               val action = FragmentADirections.actionFragmentAToFragmentB(person)
+               findNavController().navigate(action)
```
Comment (Ericgacoki) [inline app/src/main/res/duplicate/Fragment.kt]: please explain why you did this!

File: app/src/main/res/duplicate/Fragment.kt [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+           }
+
+        }?.root
```
Comment (Ericgacoki) [inline app/src/main/res/duplicate/Fragment.kt]: Root doesn't work like this!

**Review by Ericgacoki — COMMENTED** [review]
Hello @MaurineM, please go through my suggestions and make the respective corrections.

**Comment by MaurineM** [comment]
Hey @Ericgacoki, I've made the changes! Please check them out.

**Review by Ericgacoki — APPROVED** [review]
Hello @MaurineM, approved!

File: app/src/main/res/duplicate/Fragment.kt [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package com.demo2.safeargs
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.findNavController
+import com.demo.safeargs.databinding.FragmentABinding
```
Comment (Ericgacoki) [inline app/src/main/res/duplicate/Fragment.kt]: Just noticed that the View class is implemented wrongly.

**Review by Ericgacoki — CHANGES_REQUESTED** [review]

**Review by Ericgacoki — APPROVED** [review]

**Comment by Ericgacoki** [comment]
Compatible but won't merge due to some reasons. We can still reopen this later. thank you!

File: app/src/main/res/duplicate/Fragment.kt [hunk 5]
```diff
@@ -0,0 +1,31 @@
+package com.demo2.safeargs
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment as Fragment
+import androidx.navigation.fragment.findNavController
+import com.demo.safeargs.databinding.FragmentABinding
```
Comment (Ericgacoki) [inline app/src/main/res/duplicate/Fragment.kt:9]: Nice imports.

File: app/src/main/res/duplicate/Fragment.kt [hunk 6]
```diff
@@ -0,0 +1,31 @@
+package com.demo2.safeargs
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
```
Comment (Ericgacoki) [inline app/src/main/res/duplicate/Fragment.kt]:
```suggestion
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View1
```

File: app/src/main/res/duplicate/Fragment.kt (hunk 5 above)
Comment (Ericgacoki) [inline app/src/main/res/duplicate/Fragment.kt]:
```suggestion
import androidx.navigation.fragment.findNavController
import com.demo.safeargs.databinding.FragmentABinding 2
```

File: app/src/main/res/duplicate/Fragment.kt [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+import androidx.fragment.app.Fragment as Fragment
+import androidx.navigation.fragment.findNavController
+import com.demo.safeargs.databinding.FragmentABinding
+
+class FragmentA: Fragment() {
+    private var binding : FragmentABinding? = null
+    private val no = null
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
```
Comment (Ericgacoki) [inline app/src/main/res/duplicate/Fragment.kt]:
```suggestion
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?, 3
```

File: app/src/main/res/duplicate/Fragment.kt [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+           binding?.btnShareData?.setOnClickListener {
+               val person = Person("Reen", 24)
+               val action = FragmentADirections.actionFragmentAToFragmentB(person)
+               findNavController().navigate(action)
+           }
+           
+        }?.root
+        // Inflate the layout
+    }
```
Comment (Ericgacoki) [inline app/src/main/res/duplicate/Fragment.kt]:
```suggestion
        }?.root
    }
```
remove roots

**Review by Ericgacoki — CHANGES_REQUESTED** [review]

**Review by Ericgacoki — APPROVED** [review]

File: app/src/main/res/duplicate/Fragment.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+{ "_id" : ObjectId("60db758c4a433597bcae61c1"), "student_number" : 4, "name" : "John Dohn", "age" : 37, "Course" : "Political Science" }
+{ "_id" : ObjectId("60db758c4a433597bcae61c2"), "student_number" : 2, "name" : "Joan Dee", "age" : 23, "Course" : "Computer Animation" }
+{ "_id" : ObjectId("60db758c4a433597bcae61c3"), "student_number" : 3, "name" : "Jim Doeh", "age" : 19, "Course" : "Industrial Science" }
+{ "_id" : ObjectId("60db7a59ee65088f9b51677e"), "student_number" : 5, "name" : "Last student added", "age" : 22, "Course" : "Engineering" }
+```
+### Export JSON to MongoDB Database
+Now let us export the new file to JSON using the following command:
+```bash
+mongoexport --db studentinfo --collection student --out d:\out.json --pretty
+```
```
Comment (Ericgacoki) [inline app/src/main/res/duplicate/Fragment.md:37]:
```suggestion
### Export JSON to MongoDB Database (commit 1)
Now let us export the new file to JSON using the following command:
```bash
mongoexport --db studentinfo --collection student --out d:\out.json
```
```

**Review by Ericgacoki — CHANGES_REQUESTED** [review]

File: app/src/main/res/duplicate/Fragment.md [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+WriteResult({ "nInserted" : 1 })
+```
+Now if we run the command, we find an extra record in the database:
+```json
+db.students.find()
+{ "_id" : ObjectId("60db758c4a433597bcae61c0"), "student_number" : 1, "name" : "John Doe", "age" : 17, "Course" : "Computer Science" }
+{ "_id" : ObjectId("60db758c4a433597bcae61c1"), "student_number" : 4, "name" : "John Dohn", "age" : 37, "Course" : "Political Science" }
+{ "_id" : ObjectId("60db758c4a433597bcae61c2"), "student_number" : 2, "name" : "Joan Dee", "age" : 23, "Course" : "Computer Animation" }
+{ "_id" : ObjectId("60db758c4a433597bcae61c3"), "student_number" : 3, "name" : "Jim Doeh", "age" : 19, "Course" : "Industrial Science" }
+{ "_id" : ObjectId("60db7a59ee65088f9b51677e"), "student_number" : 5, "name" : "Last student added", "age" : 22, "Course" : "Engineering" }
```
Comment (Ericgacoki) [inline app/src/main/res/duplicate/Fragment.md:31]:
```suggestion
```json
db.students.find() (did time move?)
{ "_id" : ObjectId("60db758c4a433597bcae61c0"), "student_number" : 1, "name" : "John Doe", "age" : 17, "Course" : "Computer Science" }
{ "_id" : ObjectId("60db758c4a433597bcae61c1"), "student_number" : 4, "name" : "John Dohn", "age" : 37, "Course" : "Political Science" }
{ "_id" : ObjectId("60db758c4a433597bcae61c2"), "student_number" : 2, "name" : "Joan Dee", "age" : 23, "Course" : "Computer Animation" }
{ "_id" : ObjectId("60db758c4a433597bcae61c3"), "student_number" : 3, "name" : "Jim Doeh", "age" : 19, "Course" : "Industrial Science" }
{ "_id" : ObjectId("60db7a59ee65088f9b51677e"), "student_number" : 5, "name" : "Last student added", "age" : 22, "Course" : "Engineering" }
```

**Review by Ericgacoki — CHANGES_REQUESTED** [review]

File: app/src/main/res/duplicate/Fragment.md [hunk 11]
```diff
@@ -0,0 +1,33 @@
+This command should fetch all objects from the JSON file that was imported as below:
+```json
+db.students.find()
```
Comment (Ericgacoki) [inline app/src/main/res/duplicate/Fragment.md:3]:
```suggestion
This command should fetch all objects from the JSON file that was imported as below:
```json
db.students.find() is it long ago?
```

File: app/src/main/res/duplicate/Fragment.md [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+### Save JSON data to MongoDB Database
+Now that we have imported a JSON file into MongoDB, we will add two records to the database then export the data to a new JSON file.
+Use the command below to add records to the database:
+```bash
+db.students.insert({"student_number":5, "name":"Last student added", "age":22, "Course":"Engineering"})
+```
+You should get the response below in the terminal:
+```bash
+WriteResult({ "nInserted" : 1 })
+```
```
Comment (Ericgacoki) [inline app/src/main/res/duplicate/Fragment.md]: A good way to represent dependency!

**Comment by Ericgacoki** [comment]
Closing for inactivity

**Pull request closed without merging**